### ISSUE TITLE
Not all source files get loaded properly when debugging dump file

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#21267]
I download the dump file from https://github.com/dotnet/roslyn/issues/21194 and started debugging it. The first stack frame which loaded was properly shown with source (downloaded to "C:\Users\jcouv\AppData\Local\Temp\SymbolCache\src\dotnet\roslyn\6969b516a24b460e269b1ae2be09d8c4cd220c27\src\Compilers\CSharp\Portable\Binder\Binder_Symbols.cs").
But when I opened other frames in the stack, the source did not get shown.

I've used a local repo (synced to that git hash) as a workaround. I need to confirm this issue after removing the workaround, before we can investigate further.

**Comment by jcouv** [comment]
Confirmed the issue. I removed my local workaround, deleted files from the symbol cache folder, re-opened the dump file.

Some of the files that fail to open: PEMethodSymbol.cs, MetadataDecoder.cs, MethodSymbol.cs.

Some of the files that successfully open: Binder_Symbol.cs, SourceAssemblySymbol.cs, NonMissingAssemblySymbol.cs.

![image](https://user-images.githubusercontent.com/12466233/28930034-73023f96-7826-11e7-90cd-57960d2e1d49.png)

@tmat I'll stop by to discuss (how can I extract the doc table for an assembly from a dump file?).
FYI @jaredpar

**Comment by jaredpar** [comment]
Which SHA comes from that dump file?

**Comment by jcouv** [comment]
The Roslyn SHA for this dump file is 6969b516a24b460e269b1ae2be09d8c4cd220c27.

**Comment by jaredpar** [comment]
That SHA is before a PR I made to fix a number of GitLink bugs #19736.  Prior to that PR when GitLink was failing to embed source files it would exit with 0 and our build would be none the wiser. I found this out because a number of our files weren't being embedded properly. It's possible this is just one of the silent failures we saw. 

Relevant GitLink PR: https://github.com/GitTools/GitLink/pull/157
